### PR TITLE
Feature indie pharmacies

### PR DESCRIPTION
**Pull request by jknapp25**
Feature 2 indie pharmacies if they are within distance of the patient. Our list includes indie's located in and around Austin, TX.

Ticket: [Feature indie pharmacies](https://www.notion.so/photons/Feature-indie-pharmacies-0765fe4d30d34de883e249e64cb2d602?pvs=4)

### CODE REVIEW

File: apps/patient/src/components/PickupOptions.tsx [hunk 1]
```diff
@@ -10,10 +10,11 @@ import {
 
 import { PharmacyCard } from './PharmacyCard';
 import t from '../utils/text.json';
-import { Pharmacy } from '../utils/models';
+import { AUSTIN_INDIE_PHARMACY_IDS } from '../views/Pharmacy';
+import { Pharmacy as PharmacyType } from '../utils/models';
```
Comment (samkotlove) [inline apps/patient/src/components/PickupOptions.tsx:14]: Was `Pharmacy` taken?

File: apps/patient/src/views/Pharmacy.tsx [hunk 2]
```diff
@@ -54,6 +81,142 @@ const query = (method, data) =>
...
+const geocode = async (address: string) => {
+  const data = await geocoder.geocode({ address });
+  if (data?.results) {
+    return {
+      address: data.results[0].formatted_address,
+      lat: data.results[0].geometry.location.lat(),
+      lng: data.results[0].geometry.location.lng()
+    };
+  } else {
+    throw new Error('Failed to geocode the address.');
```
Comment (samkotlove) [inline apps/patient/src/views/Pharmacy.tsx:93]: Does google have an error message we can include?

File: apps/patient/src/views/Pharmacy.tsx [hunk 3]
```diff
@@ -113,127 +279,152 @@ export const Pharmacy = () => {
...
-    } catch (error) {
-      console.error(JSON.stringify(error, undefined, 2));
-      console.log(error);
-      if (error?.response.errors[0].message === 'No pharmacies found near location') {
-        setShowingAllPharmacies(true);
+    // Prepare pharmacies to show
+    let newPharmacies: PharmacyType[] = pharmaciesResult;
+
+    // Feature pharmacies
+    newPharmacies = sortIndiePharmaciesFirst(
```
Comment (samkotlove) [inline apps/patient/src/views/Pharmacy.tsx:337]: will this happen after a patient clicks "load more"? wondering if it's weird that there are indie pharmacies at the top of every click load more

File: apps/patient/src/views/Pharmacy.tsx [hunk 4]
```diff
@@ -54,6 +81,142 @@ const query = (method, data) =>
...
+    );
+    return query.pharmaciesByLocation;
+  } catch (error) {
+    if (
+      error?.response?.errors &&
+      error.response.errors[0].message === 'No pharmacies found near location'
+    ) {
+      const { message } = error.response.errors[0];
+      console.log(message);
+      return message;
```
Comment (samkotlove) [inline apps/patient/src/views/Pharmacy.tsx:136]: Don't like taht this function sometimes returns a list of pharmacies, sometimes a string, and sometimes an error

File: apps/patient/src/components/PickupOptions.tsx (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/components/PickupOptions.tsx:14]: By the component itself

File: apps/patient/src/views/Pharmacy.tsx (hunk 3 above)
Comment (samkotlove) [reply apps/patient/src/views/Pharmacy.tsx:337]: Ah looks like this is only called once, and enrichPharmacies is called for subsequent calls. Will move this comment down there

Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx:337]: nope, just on initialize(), which happens on initial load or after location change

File: apps/patient/src/views/Pharmacy.tsx [hunk 5]
```diff
@@ -113,127 +279,152 @@ export const Pharmacy = () => {
...
+
+    // Enrich existing pharmacies if we already have them
+    if (pharmacyOptions.length < fetchedPharmacies.length) {
+      const startIndex = pharmacyOptions.length;
+      const endIndex =
+        fetchedPharmacies.length - 1 < startIndex + 3
+          ? fetchedPharmacies.length - 1
+          : startIndex + 3;
+      for (let i = startIndex; i < endIndex; i++) {
+        if (totalNewEnriched === 3) break;
```
Comment (samkotlove) [inline apps/patient/src/views/Pharmacy.tsx:367]: if we show say 10, we're only showing rating/hours for 3 then?

File: apps/patient/src/views/Pharmacy.tsx [hunk 6]
```diff
@@ -113,127 +279,152 @@ export const Pharmacy = () => {
...
+
+  const handleShowMore = async () => {
+    setLoadingPharmacies(true);
+
+    const updatedOptions = [...pharmacyOptions];
+
+    let totalNewEnriched = 0;
+
+    // Enrich existing pharmacies if we already have them
+    if (pharmacyOptions.length < fetchedPharmacies.length) {
```
Comment (samkotlove) [inline apps/patient/src/views/Pharmacy.tsx:360]: Confused by the logic in here, can we get more explainations in comments?

File: apps/patient/src/views/Pharmacy.tsx (hunk 5 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx:367]: no, we only enrich 3 at a time so it doesn't take a long time to load

File: apps/patient/src/views/Pharmacy.tsx (hunk 6 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx:360]: Sorry, tried to be explicit with the comments. I'll do another pass.

File: apps/patient/src/views/Pharmacy.tsx (hunk 2 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx]: statuses, added

File: apps/patient/src/views/Pharmacy.tsx (hunk 4 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx]: ah updated

File: apps/patient/src/views/Pharmacy.tsx (hunk 6 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx:393]: k, let me know what you think!

**Review by samkotlove — APPROVED** [review]

**Review by samkotlove — APPROVED** [review]

**Pull request merged**